File: services/device/src/main/java/com/example/iter/device/dto/response/EquipmentSummaryResponse.java
package com.example.iter.device.dto.response;

import com.example.iter.device.domain.entity.EquipmentCategory;
import com.example.iter.device.domain.entity.ProductConditionType;

import java.math.BigDecimal;
import java.time.LocalDate;

public record EquipmentSummaryResponse(
        Long id,
        String name,
        EquipmentCategory category,
        BigDecimal dailyPrice,
        LocalDate availableFrom,
        LocalDate availableTo,
        ProductConditionType productCondition,
        String thumbnailUrl,
        double averageRating,
        long reviewCount
) {
}

File: services/device/src/main/kotlin/iter/device/dto/response/EquipmentSummaryResponse.kt
package iter.device.dto.response

import iter.device.domain.entity.EquipmentCategory
import iter.device.domain.entity.ProductConditionType

import java.math.BigDecimal
import java.time.LocalDate

data class EquipmentSummaryResponse(
    val id: Long,
    val name: String,
    val category: EquipmentCategory,
    val dailyPrice: BigDecimal,
    val availableFrom: LocalDate?,
    val availableTo: LocalDate?,
    val productCondition: ProductConditionType,
    val thumbnailUrl: String?,
    val averageRating: Double,
    val reviewCount: Long,
)
